pkg/netpol/eval/check.go

```diff
@@ -76,51 +76,59 @@ func (pe *PolicyEngine) convertWorkloadPeerToPodPeer(peer Peer) (*k8s.PodPeer, e
 
 // for connectivity considerations, when requesting allowed connections between 2 workload peers which are the same,
 // looking for 2 different pod instances, if exist (to avoid the trivial case of connectivity from pod to itself)
-func (pe *PolicyEngine) changePodPeerToAnotherPodObject(peer *k8s.PodPeer) {
+// returns true if different pod instances found
+func (pe *PolicyEngine) changePodPeerToAnotherPodObject(peer *k8s.PodPeer) bool {
 	// look for another pod, different from peer.Pod, with the same owner
 	for _, pod := range pe.podsMap {
 		if pod.Namespace == peer.Pod.Namespace && pod.Name != peer.Pod.Name && pod.Owner.Name == peer.Pod.Owner.Name {
 			peer.Pod = pod
-			break
+			return true
 		}
 	}
+	return false
 }
 
 // AllAllowedConnectionsBetweenWorkloadPeers returns the allowed connections from srcPeer to dstPeer,
+// and a bool flag indicating if the allowed connections are trivial
+// i.e. Connection from deployment to itself when the deployment has only one replica.
 // expecting that srcPeer and dstPeer are in level of workloads (WorkloadPeer)
-func (pe *PolicyEngine) AllAllowedConnectionsBetweenWorkloadPeers(srcPeer, dstPeer Peer) (Connection, error) {
+func (pe *PolicyEngine) AllAllowedConnectionsBetweenWorkloadPeers(srcPeer, dstPeer Peer) (Connection, bool, error) {
+	isTrivial := false
```

Comment:
```suggestion
```
move the definition of is`Trivial` to be before the call to `changePodPeerToAnotherPodObject` , where it is relevant